### PR TITLE
Favorites endpoint

### LINKED ISSUE #35: Add favorites endpoints

### PR DESCRIPTION
**Pull request by DanielMulitauopele**
- [X] Wrote Tests
- [X] Implemented
- [X] Reviewed

# Necessary checkmarks
- [X] All Tests are Passing
- [X] The code will run locally

## Type of change
- [X] New feature
- [ ] Bug Fix

# Implements/Fixes
* Adds favorites endpoints: GET all favorites and POST to favorites. The controller actions are interesting in that they can't pull the user id other than through the @current_user variable that is set in the application controller (props to Autumn for thinking of it). In this way though, we don't need to expose the user's id when pulling their favorites.

closes #35 

# Check the correct boxes

- [X] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

# Testing Changes
- [ ] No Tests have been changed
- [X] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

# Checklist

- [ ] My code has no unused/commented out code
- [X] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have fully tested my code

### CODE REVIEW

File: spec/requests/favorite_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
+require 'rails_helper'
+
+describe 'Favorites API' do
+  describe 'valid request with auth' do
+    before (:each) do
+      user_params = { email: "groot@email.com",
```
Comment (zingsurfer) [inline spec/requests/favorite_request_spec.rb:6]: Groot is the best!!

File: db/schema.rb [hunk 2]
```diff
@@ -26,9 +26,18 @@
     t.string "percent_change_24_hr"
```
Comment (zingsurfer) [inline db/schema.rb]: Are we using this assets table anywhere? If not, our new favorites table could replace it.

**Review by zingsurfer — CHANGES_REQUESTED** [review]
This looks awesome! One thing I'd recommend is either deleting the assets table if we aren't using it, or removing the favorites table attributes `price_usd` and `percent_change_24_hour`. They are duplicated in this table and our assets table, and this generally is not recommended. What do you think? It looks to me like we're not using it, so let's just take it out if that's the case. Just double check that the tests still pass, and everything still works.

File: db/schema.rb (hunk 2 above)
Comment (DanielMulitauopele) [reply db/schema.rb]: We are not! I'll drop and see how testing goes.

Comment (DanielMulitauopele) [reply db/schema.rb]: Dropped it and all pass!

**Review by DanielMulitauopele — COMMENTED** [review]
Fixed I think!

**Comment by zingsurfer** [comment]
Thanks Daniel, looks like it's good to go! I'm glad to see all of our Travis checks are passing!

**Review by zingsurfer — COMMENTED** [review]
Good call! These changes look great.

**Pull request merged**